src/Microsoft.Azure.WebJobs.Host/Executors/StorageAccountProviderExtensions.cs

```diff
@@ -22,14 +22,19 @@ public static Task<IStorageAccount> GetDashboardAccountAsync(this IStorageAccoun
             return provider.GetAccountAsync(ConnectionStringNames.Dashboard, cancellationToken);
         }
 
-        public static async Task<IStorageAccount> GetStorageAccountAsync(this IStorageAccountProvider provider, CancellationToken cancellationToken)
+        public static async Task<IStorageAccount> GetStorageAccountAsync(this IStorageAccountProvider provider, CancellationToken cancellationToken, string connectionStringName = null)
```

Comment:
How about we rename IStorageAccountProvider.GetAccountAsync to GetAccountOrNullAsync since it can return null. Then we won't have name collisions with our extension methods that validate and throw. You can then leave this method signature as is, and add a new extension method GetAccountAsync that takes the connection name and validates/throws. You'll then have this method and the one below call GetAccountAsync to do the validation and return the account.

The one place where we allow null for Dashboard account, you'll then use the GetAccountOrNull method. All other places are doing validation.l